### ISSUE TITLE
Existing Standards

### ISSUE THREAD

**Issue by oeed** [oeed/CraftOS-Standards#19]
To get this repo in a state where it is truely useable we need to have as many of the existing standards/formats/whatevers submitted. If you can think of something missing comment below or make a PR!

**Comment by demhydraz** [comment]
Diffing?

**Comment by viluon** [comment]
Diffs and patch functionality should be definitely standardized!

**Comment by demhydraz** [comment]
Woop. I'm working on implementing `diff -u`  right now.

**Comment by oeed** [comment]
Didn't we already sorta discuss diffs in #1?

**Comment by demhydraz** [comment]
Uhm, no. I mean unix diffs, the kind generated by the `diff -u` program.

**Comment by oeed** [comment]
...being?

**Comment by demhydraz** [comment]
Aren't you on OS X? go try it out for heaven's sake

**Comment by oeed** [comment]
Don't really get what it has to do with standards though.. does it save the difference to a file?

**Comment by viluon** [comment]
@oeed it's the generated difference file displayed when viewing [a commit](https://github.com/oeed/CraftOS-Standards/commit/a3d45c028b0603acbdee640fc765dc90378e66d0) which shows which lines were added/removed. You can also see it when you edit a file in Sublime, right click and click on "Show unsaved changes".

**Comment by ghost** [comment]
.nbs noteblock songs?

**Comment by viluon** [comment]
Could you link us to software using `.nbs` please @justsomeoneelseagain ?

**Comment by ghost** [comment]
www.computercraft.info/forums2/index.php?/topic/2384-noteblock-api-noteblock-player/

**Comment by viluon** [comment]
[NoteBlock API](http://www.computercraft.info/forums2/index.php?/topic/2384-noteblock-api-noteblock-player/) was released on 3rd July 2012. While it might very well work on current versions of ComputerCraft, I doubt it will. I didn't look into the code but I suspect it uses coloured cables, which are deprecated now.

**Comment by ghost** [comment]
i just searched for a noteblock image format

**Comment by ghost** [comment]
i mean audio format

**Comment by ghost** [comment]
maybe @lyqyd ´s packman repository format if he agrees

**Comment by viluon** [comment]
His nickname is @lyqyd @justsomeoneelseagain :wink:

**Comment by ghost** [comment]
yes, I realyced it afterwards

**Comment by lyqyd** [comment]
Yeah, I'd have no problem with that.  I don't really have the time/energy to write it up, but I'd be happy to answer any questions on it that the readme in the packman repository can't.

**Comment by Luca0208** [comment]
we might want to standardize the intendation(Is that a word?) 2 spaces might be good, as that is what the normal edit program uses, someone can change Lua.md if enough people agree

**Comment by oeed** [comment]
@Luca0208 Indentation tends to be fair subjective. Some people like tabs, others spaces. 2 spaces works better on the CC screens because the screen size isn't very big, but personally, I find 4 spaces much easier to read in external editors and only use that.

Hence, I'm not sure whether it's worth standardising. It's not something that's causing clashes or anything like that.

**Comment by Luca0208** [comment]
The big problem comes when someone uses tabs to intend. It just looks awful in CC

**Comment by viluon** [comment]
@Luca0208 that is the problem of CC, not tab indent. Few actual developers use the flawed ingame editor. Still not a reason to standardize indentation.

**Comment by demhydraz** [comment]
Indent with hard tabs, align with spaces. Fuck the in game editor.

**Comment by lyqyd** [comment]
Indentation is personal choice, and outside the scope of this project. The standards should be about how programs work or interact with other code/files, not about how they look.

**Comment by viluon** [comment]
Agreed @lyqyd (and actually even with @demhydraz ).

**Comment by viluon** [comment]
Oops, this isn't a proposal, sorry!

**Comment by CrazedProgrammer** [comment]
What about .skch (layered image file from Oeed's Sketch) and .nfa (animation file from NitrogenFingers' NPaintPro)  files?
I consider them pretty standard and useful.

**Comment by oeed** [comment]
Yeah I don't object to that, sketch files and nfa do seem to be used a bit.

**Comment by lyqyd** [comment]
#59 contains a proposal for adding (a slightly modified version of) TRoR.
